docs/the-new-architecture/pillars-turbomodule.md

```diff
@@ -3,16 +3,699 @@ id: pillars-turbomodules
+          package="com.RTNCalculator">
+</manifest>
+```
+
+This is a small manifest file that defines the package for our module.
+
+
+##### CalculatorModule.java
+
+```java title="CalculatorModule.java"
+package com.RTNCalculator;
+
+import androidx.annotation.NonNull;
+import com.facebook.react.bridge.NativeModule;
+import com.facebook.react.bridge.Promise;
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.bridge.ReactContext;
+import com.facebook.react.bridge.ReactContextBaseJavaModule;
+import com.facebook.react.bridge.ReactMethod;
+import java.util.Map;
+import java.util.HashMap;
+
+public class CalculatorModule extends NativeCalculatorSpec {
+
+    CalculatorModule(ReactApplicationContext context) {
+        super(context);
+    }
+
+    @Override
+    @NonNull
+    public String getName() {
+        return "RTNCalculator";
+    }
+
+    @Override
+    public void add(double a, double b, Promise promise) {
+        promise.resolve(a + b);
+    }
+}
+```
+
+This class implements the module itself, which extends the `NativeCalculatorSpec` that was generated from the `NativeCalculator` JavaScript specification file.
+
+##### CalculatorPackage.java
+
+```java title="CalculatorPackage.java"
+package com.RTNCalculator;
+
+import androidx.annotation.Nullable;
+import com.facebook.react.bridge.NativeModule;
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.module.model.ReactModuleInfo;
+import com.facebook.react.module.model.ReactModuleInfoProvider;
+import com.facebook.react.TurboReactPackage;
+import com.facebook.react.uimanager.ViewManager;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.HashMap;
+import java.util.Map;
+
+public class CalculatorPackage extends TurboReactPackage {
+
+  @Nullable
+  @Override
+  public NativeModule getModule(String name, ReactApplicationContext reactContext) {
+      if (name.equals("RTNCalculator")) {
+          return new CalculatorModule(reactContext);
+      } else {
+          return null;
+      }
+  }
+
+
+  @Override
+  public ReactModuleInfoProvider getReactModuleInfoProvider() {
+      return () -> {
+          final Map<String, ReactModuleInfo> moduleInfos = new HashMap<>();
+          moduleInfos.put(
+                  "RTNCalculator",
+                  new ReactModuleInfo(
+                          "RTNCalculator",
+                          "RTNCalculator",
+                          false, // canOverrideExistingModule
+                          false, // needsEagerInit
+                          true, // hasConstants
+                          false, // isCxxModule
+                          false // isTurboModule
+          ));
+          return moduleInfos;
+      };
+  }
+
+}
+```
+
+This is the last piece of Native Code for Android. It defines the Package object that will be used by the app to load the module.
+
+## 5. Adding the Fabric Component To Your App
```

Comment:
```suggestion
## 5. Adding the TurboModule To Your App
```

Reply:
addressed, thanks for catching that 🤦‍♀️